### PR TITLE
Avoid $a, $b

### PR DESCRIPTION
**Pull request by grahamc**
Implementing part of #1003: 

> Using $a or $b outside sort() at line 45, column 16.  $a and $b are special package variables for use in sort() and related functions. Declaring them as lexicals like "my $a" may break sort(). Use different variable names.  (Severity: 4)

Care should be taken to verify each commit doesn't stomp on existing variables in scope.

### CODE REVIEW

File: src/lib/Hydra/Plugin/BitBucketStatus.pm [hunk 1]
```diff
@@ -22,21 +22,21 @@ sub toBitBucketState {
...
-        my $evals = $build->jobsetevals;
+    foreach my $build ($topbuild, @{$dependents}) {
+        my $jobName = showJobName $build;
+        my $evals = $topbuild->jobsetevals;
         my $ua = LWP::UserAgent->new();
         my $body = encode_json(
             {
-                state => $finished ? toBitBucketState($b->buildstatus) : "INPROGRESS",
-                url => "$baseurl/build/" . $b->id,
+                state => $finished ? toBitBucketState($tbuild->buildstatus) : "INPROGRESS",
```
Comment (Mindavi) [inline src/lib/Hydra/Plugin/BitBucketStatus.pm:35]: I assume this must be $build? Same for the other 3 below.

Comment (grahamc) [reply src/lib/Hydra/Plugin/BitBucketStatus.pm:35]:
```suggestion
                state => $finished ? toBitBucketState($build->buildstatus) : "INPROGRESS",
```

File: src/lib/Hydra/Plugin/BitBucketStatus.pm [hunk 2]
```diff
@@ -22,21 +22,21 @@ sub toBitBucketState {
...
+                url => "$baseurl/build/" . $tbuild->id,
```
Comment (grahamc) [inline src/lib/Hydra/Plugin/BitBucketStatus.pm:36]:
```suggestion
                url => "$baseurl/build/" . $build->id,
```

File: src/lib/Hydra/Plugin/BitBucketStatus.pm [hunk 3]
```diff
@@ -22,21 +22,21 @@ sub toBitBucketState {
...
                 name => $jobName,
-                key => $b->id,
-                description => "Hydra build #" . $b->id . " of $jobName",
+                key => $tbuild->id,
```
Comment (grahamc) [inline src/lib/Hydra/Plugin/BitBucketStatus.pm:38]:
```suggestion
                key => $build->id,
```

File: src/lib/Hydra/Plugin/BitBucketStatus.pm [hunk 4]
```diff
@@ -22,21 +22,21 @@ sub toBitBucketState {
...
+                description => "Hydra build #" . $tbuild->id . " of $jobName",
```
Comment (grahamc) [inline src/lib/Hydra/Plugin/BitBucketStatus.pm:39]:
```suggestion
                description => "Hydra build #" . $build->id . " of $jobName",
```

**Comment by grahamc** [comment]
I'm pretty sure this is ready to go.

**Pull request merged**